### DISCUSSION TITLE
Ruby 3.3.0 install fails building psych yaml.h not found

### DISCUSSION THREAD

**Discussion by whitehome** [rbenv/ruby-build#2334] [Build failures] [answered]
Ruby 3.3.0 installation fails on Rocky Linux 9.3 building psych.

yaml.h is not found but exists in location:

/usr/include/yaml-cpp/yaml.h

Installed yaml packages below.  Unable to install yaml-dev package because it doesn't exist.

yaml-cpp.x86_64                                          0.6.3-5.el9                                       @epel        
yaml-cpp-devel.x86_64                                    0.6.3-5.el9                                       @epel  

Installed.

Log file shows yaml.h was not found during verification then. Fails during psych build:

configuring psych
yaml.h not found

*** Following extensions are not compiled:
psych:
        Could not be configured. It will not be installed.
        Check ext/psych/mkmf.log for more details.
*** Fix the problems, then remove these directories and try again if you want.
make[1]: *** [exts.mk:2422: note] Error 1
make[1]: Leaving directory '/tmp/ruby-build.20240111082130.142352.zsAdQ1/ruby-3.3.0'
make: *** [uncommon.mk:398: build-ext] Error 2
external command failed with status 2

I did manually add yaml.h to global includes and yaml.h was found but then ios was missing.  Can you help me?

**Comment by whitehome** [comment]
Didn't list libyaml installation:
libyaml.x86_64                                             0.2.5-7.el9                                          @minimal

**Reply by whitehome** [reply, marked as answer]
Solved my issue in the follow way:

1. Disabled rbenv
2. Installed ruby-install
3. ruby-install needed libyamal-devel and gdmb-devel to continue (note id did that up front!)
4. As I already new libyaml-devel would not install on Rocky Linux 9 with dnf I finally foond the way to do it and gdmb-devel
5. https://rockylinux.pkgs.org/9/rockylinux-devel-x86_64/libyaml-devel-0.2.5-7.el9.x86_64.rpm.html
6. https://rockylinux.pkgs.org/9/rockylinux-crb-x86_64/gdbm-devel-1.19-4.el9.x86_64.rpm.html
7. Both installed using the instructions under install
8. I reenabled rbenv to test it and the install worked. so issue closed.
9. I'd say ruby-install did a better job of identifying what I needed to do as I could find the missing header but rbenv or reather the make file could not sounds like a LD_INCLUDE issue.

**Comment by shqear93** [comment]
```
brew info libyaml
brew list libyaml
gem install psych -- --with-libyaml-dir=/opt/homebrew/Cellar/libyaml/0.2.5
```

* don't forget to update the libyaml path

**Reply by sharikovvladislav** [reply]
The `brew list libyaml` gives me this error:
```
Error: No such keg: /opt/homebrew/Cellar/libyaml
```

**Reply by baoqchau** [reply]
In case anyone run into the error above, `brew install libyaml` should resolve the issue

**Comment by hwiorn** [comment]
On Rocky 9+, you have to enable `CRB`(=`powertools` on RHEL) repo. Then, you can install `libyaml-devel` package.

```sh
sudo dnf config-manager --set-enabled crb
sudo dnf install -y libyaml-devel
```
And, Build and install the Ruby 3.3.x.

**Comment by negrusti** [comment]
On Ubuntu:  
`apt install libyaml-dev`

**Comment by alexpapworth** [comment]
You'll want `apk add yaml-dev` if you're using Alpine 🙂
